clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackSecurityGroupCachingAgent.groovy

```diff
@@ -60,4 +80,70 @@ class OpenstackSecurityGroupCachingAgent extends AbstractOpenstackCachingAgent {
     log.info("Caching ${cacheResultBuilder.namespace(SECURITY_GROUPS.ns).keepSize()} items in ${agentType}")
     cacheResultBuilder.build()
   }
+
+  /**
+   * Build a security group rule the references another security group.
+   *
+   * This will Look up the referenced security group by name in any region. This lookup may fail if the referenced
+   * security group has not been cached yet. Due to the async nature of the caching, this is not a failure. In
+   * this case, the referenced security group will include the name, but not the id or region.
+   */
+  private SecurityGroupRule buildSecurityGroupRule(SecGroupExtension.Rule rule, ProviderCache providerCache) {
+
+    String key = Keys.getSecurityGroupKey(rule.group.name, '*', accountName, '*')
+    Collection<String> identifiers = providerCache.filterIdentifiers(SECURITY_GROUPS.ns, key)
+    Collection<CacheData> securityGroupData = providerCache.getAll(SECURITY_GROUPS.ns, identifiers)
+
+    String id = null
+    String region = null
+    if (securityGroupData.isEmpty()) {
+      // Expected if not all security groups have been cached yet, like at startup
+      log.debug("Could not find any security groups by name ${rule.group.name} in account ${accountName}")
+    } else if (securityGroupData.size() > 1) {
+      log.warn("Found too many security groups by name ${rule.group.name} in account ${accountName}")
+    } else {
+      Map<String, Object> attributes = securityGroupData.first().attributes
+      id = attributes?.id
+      region = attributes?.region
+    }
+
```

Comment:
Would it be better to call ```client.allRegions.each { ... get sec groups ... }``` instead of search the cache for potentially stale data?